x/programs/cmd/simulator/src/lib.rs

```diff
@@ -183,7 +183,7 @@ impl<'a> Plan<'a> {
 #[derive(Debug, Deserialize)]
 pub struct BaseResponse {
     /// The numeric id of the step.
-    pub id: u32,
+    pub id: usize,
```

Comment:
I think that this one should be of type `Id`, that requires having to "override" the current deserialize, I did it in https://github.com/ava-labs/hypersdk/pull/1002/files#diff-33034c207f9554212c375b13284c24b46f009907280babee29791aa8c0ec11baR205 
We should also remove the custom `Serialize` `Deserialize` implementation once we fix https://github.com/ava-labs/hypersdk/issues/1004

Reply:
I'm not going to change this then since you already did in your PR

Reply:
fair!